modules/ROOT/pages/dataweave-maven-plugin.adoc

```diff
@@ -0,0 +1,338 @@
+
+|`runnerSystemProperties`
+|`Map`
+|`null`
+|A set of additional system properties to pass to the test runner process.
+
+|`runnerArgLine`
+|`boolean`
+|`null`
+|Set an additional JVM options to pass to the test runner process
+
+|`runnerJvmDebug`
+|`boolean`
+|`false`
+|Enables the remote debug for the test runner JVM when set to `true`.
+|===
+
+=== Test Configuration Example
+
+The following example enables `sonar` coverage report and disables `htmlReport` in a project.
+
+.Example `pom.xml` file
+[source,xml,linenums]
+----
+<plugin>
+  <groupId>org.mule.weave</groupId>
+  <artifactId>data-weave-maven-plugin</artifactId>
+  <version>0.8.0-SNAPSHOT</version>
+  <extensions>true</extensions>
+  <configuration>
+    ...
+    <tests>
+      <htmlReport>false</htmlReport>
+      <coverageEnabled>true</coverageEnabled>
+      <coverageFormat>sonar</coverageFormat>
+    </tests>
+    ...
+  </configuration>
+</plugin>
+----
+
+== Generate Documentation for a DataWeave Library
+
+Run the `data-weave:generate-docs` goal to auto-generate the documentation for your DataWeave library, for example:
+
+[source,console,linenums]
+----
+mvn prepare-package
+----
+
+Alternatively, you can also run:
+
+[source,console,linenums]
+----
+mvn data-weave:generate-docs
+----
+
+=== Configure the Automated Docs Generation
+
+Inside the `plugin` configuration, add the `docs` element with your desired documentation configuration.
+
+*Configuration Parameters*
+
+[%header%autowidth.spread,cols=".^a,.^a,.^a,.^a"]
+|===
+|Name |Type |Default | Description
+
+|`output`
+|`String`
+|`${project.build.directory}/data-weave-docs`
+|The output directory where the documentation files will create.
+
+|`template`
+|`String`
+|`exchange_markdown`
+|The template set to use for the auto-generate documentation.
+
+DataWeave Maven plugin supports 3 kinds of documentation templates:
+
+* `exchange_markdown` +
+Generates the documentation in markdown format compliant with https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/exchange-experience-api/[Exchange].
+* `markdown` +
+Generates the documentation in markdown format.
+* `asciidoc` +
+Generates the documenation in asciidoc format.
+
+Valid values are: `exchange_markdown` or `markdown` or `asciidoc`.
+
+|`homePage`
+|`String`
+|`null`
+|The path to a custom markdown file to be added in the home page generated by `markdown` or `exchange_markdown` template.
+
+|`favicon`
+|`String`
+|`null`
+|Path to the image file (png/jpg) for asset's icon in the Exchange portal.
+|===
+
+=== Docs Generation Example
+
+The following example shows how to configure a custom home page and a specific favicon to include in the Exchange portal home page for your DataWeave library:
+
+.Example `pom.xml` file:
+[source,xml,linenums]
+----
+<plugin>
+  <groupId>org.mule.weave</groupId>
+  <artifactId>data-weave-maven-plugin</artifactId>
+  <version>0.8.0-SNAPSHOT</version>
```

Comment:
Do not use SNAPSHOT version.
Today, the latest data-weave-maven-plugin released version is 0.1.0 but we need to update it after this sprint end to include the latest changes.